src/pages/entropyInput.tsx

```diff
@@ -40,41 +42,55 @@ const EntropyInputPage = () => {
   )
   const handleSubmit = () => {
     if (percentage !== 100) return
-    saveGeneratedEntropy()
+    processGeneratedEntropy()
     if (provider === 'Ethereum') {
       navigate(ROUTES.DOUBLE_SIGN)
     } else {
       navigate(ROUTES.LOBBY)
     }
   }
 
+
   const handleCaptureMouseEntropy: MouseEventHandler<HTMLDivElement> = (e) => {
-    const d = new Date()
-    setMouseEntropy(
-      `${mouseEntropy}${e.movementX}${e.movementY}${e.screenX}${
-        e.screenY
-      }${d.getTime()}`
-    )
+    /*
+    Mouse entropy is based off of recording the position and precise timing of mouse movements.
+    Entropy is only collected every ~128 ms to help reduce correlated mouse locations. The precise period is sampled
+    from ~ U[0, 255] to increase sample periodicity variance.
+    performance.now() is used for timestamps due to its sub-millisecond resolution in some browsers.
+    */
+   const randOffset = randomBytes(1)[0]
```

Comment:
it might be better not to generate `randomOffset` every time mousemove event fires to increase periodicity variance. instead, what about generating `randomOffset` inside the if clause and keeping the randomOffset value as a state by using `useState` for the next entropy update?

```ts
/// in a component
const [randomOffset, setRandomOffset] = useState(0)

/// in the mouse event handler

    if (performance.now() - lastMouseEntropyUpdate > randOffset) {
      setLastMouseEntropyUpdate(performance.now())
      setMouseEntropySamples(mouseEntropySamples + 1)
      setMouseEntropy(`${mouseEntropy}${e.movementX}${e.movementY}${e.screenX}${e.screenY}${performance.now()}`)
      setRandomOffset(randomBytes(1)[0])
    }

```

Reply:
Oh that's interesting! This is definitely an improvement, both from a safety & efficiency standpoint.

Without your fix, the randomness is resampled on every mouse event meaning that it will be heavily biased towards the lower numbers, essentially eliminating the benefit of the random offset. I'll implement your suggestion now.

Reply:
Implemented in https://github.com/zkparty/trusted-setup-frontend/pull/31/commits/cccdb39e236ad397a627f86c507b66ec901436da.